opentech/apply/projects/views.py

```diff
@@ -1,30 +1,124 @@
 from copy import copy
 
 from django.contrib import messages
+from django.contrib.auth.decorators import login_required
 from django.core.exceptions import PermissionDenied
 from django.db import transaction
-from django.shortcuts import redirect
+from django.shortcuts import get_object_or_404, redirect
 from django.utils.decorators import method_decorator
 from django.utils.translation import ugettext_lazy as _
 from django.views.generic import CreateView, DetailView, FormView, UpdateView
 
 from opentech.apply.activity.messaging import MESSAGES, messenger
 from opentech.apply.activity.views import ActivityContextMixin, CommentFormView
 from opentech.apply.users.decorators import staff_required
-from opentech.apply.utils.views import (DelegateableView, DelegatedViewMixin,
-                                        ViewDispatcher)
+from opentech.apply.utils.views import (
+    DelegateableView,
+    DelegatedViewMixin,
+    ViewDispatcher
+)
 
 from .forms import (
+    ApproveContractForm,
     CreateApprovalForm,
     ProjectApprovalForm,
     ProjectEditForm,
     RejectionForm,
     RemoveDocumentForm,
     SetPendingForm,
     UpdateProjectLeadForm,
-    UploadDocumentForm,
+    UploadContractForm,
+    UploadDocumentForm
+)
+from .models import (
+    CONTRACTING,
+    IN_PROGRESS,
+    Approval,
+    Contract,
+    PacketFile,
+    Project
 )
-from .models import CONTRACTING, Approval, Project, PacketFile
+
+
+class ContractsMixin:
+    def get_context_data(self, **kwargs):
+        contracts = self.get_contracts()
+
+        try:
+            latest_contract = contracts[0]
+        except TypeError:
+            latest_contract = None
+
+        context = super().get_context_data(**kwargs)
+        context['latest_contract'] = latest_contract
+        context['contracts'] = contracts
+        return context
+
+    def get_contracts(self):
+        """
+        Get approved AND signed contracts for the current project
```

Comment:
I've had a chat with Tom and he doesn't think that listing unapproved contracts make sense, especially when we aren't allowing them to be approved through the UI? 

An example would be that the user didn't scan the document very well and needs to resubmit. in that case. We are no longer interested in the old contract after the user submits a new contract. Alternatively if OTF upload a newer unsigned contract because the terms change the unapproved signed contract is no longer valid anyway. 

so the logic should be 
* List all contracts that are both signed and approved
* Show the newest unsigned or (signed and unapproved) contract if it is newer than any contract in the approved queryset.